Src/FluentAssertions/Execution/AssertionScope.cs

```diff
@@ -279,12 +289,7 @@ public Continuation FailWith(string message, params Func<object>[] argProviders)
 
         private string GetIdentifier()
         {
-            if (!string.IsNullOrEmpty(Context))
-            {
-                return Context;
-            }
-
-            return CallerIdentifier.DetermineCallerIdentity();
+            return Context?.Value ?? CallerIdentifier.DetermineCallerIdentity();
```

Comment:
Are you sure that `Context.Value` will never return an empty string?
Before this was expected and handled by using `string.IsNullOrEmpty`.

Reply:
Well, all the tests pass. *shrug*

I can change this to match the old semantics more exactly.

Reply:
Maybe @jnyrup or @dennisdoomen have more information!?

Reply:
This might be okay, because `Context` as a `Lazy<string>` is newly-introduced, so it can simply be a convention that it never return an empty string.

Reply:
I agree that it should never contain an empty string, but I'm not sure we can guarantee that. Somebody could use the `AssertionScope` constructor and pass an empty string. So I think accounting for that is the smart thing to do.

Reply:
It is done. :-)